route-rs-runtime/src/link/mod.rs

```diff
@@ -1,14 +1,29 @@
+//! # What are they for?
+//!
+//! Links are an abstraction used by the runtime to link processors and classifiers together, and manage the flow of packets through the router.
+//! Processors and Classifiers, which implement all the non-flow business logic of the router, are loaded into links to create specfic behavior.
+//! Links are joined together via their interfaces, and the links are then dumped into a runtime to begin pulling packets through the router.
+//! When the library user uses Graphgen, Links are declared, connected, and placed into a runtime via generated code. The user does this by laying
+//! out their desired graph of Processors and Classifiers in a graph file, and Graphgen does all the work from there! Additionally, since all links
+//! must implement the LinkBuilder trait, Links are transparently composable at no cost during runtime! The most basic links that implement Poll,
+//! Stream and Future are Primitives. Links that are made by wrapping a collection of Links are called Composites. Link composition is a great way
+//! to reduce the complexity of your router, and make its design more inspectable. Users of the library are encouraged to create their own Composite
+//! Links, but should refrain from defining new Primitive Links. This prevents the user from having to worry about the complexities of generically
```

Comment:
> ...should refrain from defining new Primitive Links.

We can remove this sentence if we disallow users from adding links to the same Tokio instance?

Reply:
Hmm, I guess if we control Tokio? We haven't quite nailed down the interface yet. I agree though this is confusing, I will remove it. I can bury this somewhere inside the Primitive page instead talking about how users should not need to define their own, since graphgen will not know about then among other things.